testing/trino-testing/src/main/java/io/trino/testing/AbstractTestJoinQueries.java

```diff
@@ -2323,23 +2324,25 @@ public void testOutputDuplicatesInsensitiveJoin()
 
     private void assertJoinOutputPositions(@Language("SQL") String sql, int expectedJoinOutputPositions)
     {
-        ResultWithQueryId<MaterializedResult> result = getDistributedQueryRunner().executeWithQueryId(
-                Session.builder(getSession())
-                        .setSystemProperty(JOIN_REORDERING_STRATEGY, "NONE")
-                        .build(),
-                sql);
-        assertEquals(result.getResult().getMaterializedRows().get(0).getField(0), 0L);
-        QueryStats stats = getDistributedQueryRunner()
-                .getCoordinator()
-                .getQueryManager()
-                .getFullQueryInfo(result.getQueryId())
-                .getQueryStats();
-        int actualJoinOutputPositions = stats.getOperatorSummaries()
-                .stream()
-                .filter(summary -> summary.getOperatorType().equals("LookupJoinOperator"))
-                .map(OperatorStats::getOutputPositions)
-                .mapToInt(Math::toIntExact)
-                .sum();
-        assertEquals(actualJoinOutputPositions, expectedJoinOutputPositions);
+        assertEventually(() -> {
```

Comment:
We usually use `@Flaky(issue = "https://github.com/trinodb/trino/issues/5172",` for this.

if not that, then in the minimum, link to the issue and make it clear, you're working something around, with a TODO comment